google-cloud-storage/src/main/java/com/google/cloud/storage/GrpcConversions.java

```diff
@@ -263,52 +272,179 @@ private Bucket bucketInfoEncode(BucketInfo from) {
     to.setVersioning(versioningBuilder.build());
     ifNonNull(from.getDefaultEventBasedHold(), to::setDefaultEventBasedHold);
     ifNonNull(from.getLabels(), to::putAllLabels);
-    // preserve mapping to deprecated property
-    Bucket.Lifecycle.Builder lifecycleBuilder = Bucket.Lifecycle.newBuilder();
+    to.setLifecycle(buildLifecyclePolicy(from));
+    ifNonNull(from.getLogging(), loggingCodec::encode, to::setLogging);
+    ifNonNull(from.getCors(), toImmutableListOf(cors()::encode), to::addAllCors);
     ifNonNull(
-        from.getLifecycleRules(),
-        Utils.toImmutableListOf(lifecycleRule()::encode),
-        lifecycleBuilder::addAllRule);
-    /*
-    // Corruption occurs by having deleteRule() values added
+        from.getOwner(),
+        lift(entity()::encode).andThen(o -> Owner.newBuilder().setEntity(o).build()),
+        to::setOwner);
     ifNonNull(
-       from.getDeleteRules(),
-       Utils.toImmutableListOf(deleteRule()::encode),
-       lifecycleBuilder::addAllRule);
-    */
-    to.setLifecycle(lifecycleBuilder.build());
-    // TODO(frankyn): Add logging decoder
-    // TODO(frankyn): Add entity decoder
-    // TODO(frankyn): Add Cors decoder support
-    // TODO(frnakyn): Add DefaultObjectAcl decoder support
-    // TODO(frankyn): Add lifeycle decoder
-    // TODO(frankyn): Add deleteRules decoder
+        from.getDefaultAcl(), toImmutableListOf(objectAcl()::encode), to::addAllDefaultObjectAcl);
+    ifNonNull(from.getIamConfiguration(), iamConfiguration()::encode, to::setIamConfig);
     // TODO(frankyn): Add SelfLink when the field is available
-    // TODO(frankyn): Add decoder for iamConfig
     // TODO(frankyn): Add Etag when support is avialable
     return to.build();
   }
 
+  private Bucket.Lifecycle buildLifecyclePolicy(BucketInfo from) {
+    // Handle duplicate rules introduced by deleteRules using a backing ImmutableSet
```

Comment:
Handling duplicates, what are your thoughts @BenWhitehead; I'm deduping proto Rule message types, but not sure if equals() will handle unknown fields as different. Looking into this next; but thought you may already know.

Reply:
Looks like this should be okay, tested an unknown field on Rule (thankfully action and condition are both messages as well): Let me know if I goofed.

```
  @Test
  public void assertNotEqualWithUnknownField() {
    com.google.storage.v2.Bucket.Lifecycle.Rule rule =
        com.google.storage.v2.Bucket.Lifecycle.Rule.getDefaultInstance();

    assertEquals(
        com.google.storage.v2.Bucket.Lifecycle.Rule.getDefaultInstance()
            .toBuilder()
            .setUnknownFields(
                UnknownFieldSet.newBuilder()
                    .addField(100, UnknownFieldSet.Field.newBuilder().addFixed32(10).build())
                    .build()).build(), com.google.storage.v2.Bucket.Lifecycle.Rule.getDefaultInstance());
  }
```

Reply:
Hmm, duplicates is an interesting case that we haven't thus far accounted for.

In fact the existing BucketInfo.fromPb[1] is duplicating and not deduplicating later.

I think we're going to want to start another PR after merging this one, where we actually go through and remove all internal usage of DeleteRule and instead translate the respective delete rule operations into the correct operations for lifecycle rules. Then, we can remove this duplication from both the apiary and grpc converters.

Regarding handling of unknown fields, we only have to account for them when decoding to BlobInfo. Once the BlobInfo is decoded, the unknown fields will effectively be abandoned. 

[1] https://github.com/googleapis/java-storage/blob/main/google-cloud-storage/src/main/java/com/google/cloud/storage/BucketInfo.java#L2200-L2219

Reply:
We spoke offline, we're going to open a separate PR to remove the duplication in BucketInfo and instead shim the old DeleteRules over exclusively LifecycleRules